packages/viem/src/utils/interop/calculateChecksum.ts

```diff
@@ -0,0 +1,42 @@
+import { encodePacked, keccak256, maxUint32, maxUint64, toHex } from 'viem'
+
+import type { MessageIdentifier } from '@/types/interop/executingMessage.js'
+
+/**
+ * Calculates the checksum for a cross chain message following the same logic as CrossL2Inbox.sol
+ */
+export function calculateChecksum(
+  id: MessageIdentifier,
+  msgHash: `0x${string}`,
+) {
```

Comment:
nit on being explicit with return types
```suggestion
): Hex {
```